httplog/httplog.go

```diff
@@ -48,25 +49,31 @@ func (l *Logger) LogFunc() middleware.Logger {
 	switch l.mode {
 	case "", None:
 		return func(e middleware.LogEntry) {}
+	case Route:
+		return func(e middleware.LogEntry) {
+			var w logWriter
+			w.RouteLine(e)
+			l.log("%s", w.String())
+		}
 	case URL:
 		return func(e middleware.LogEntry) {
 			var w logWriter
-			w.Line(e)
-			l.log(w.String())
+			w.URLLine(e)
+			l.log("%s", w.String())
 		}
 	case Headers:
 		return func(e middleware.LogEntry) {
 			var w logWriter
-			w.Line(e)
+			w.RouteLine(e)
 			w.Dump(e)
-			l.log(w.String())
+			l.log("%s", w.String())
```

Comment:
I'm curious why this line changed - will it now be lazily evaluated?

Reply:
It's like calling `fmt.Printf("%s", "text")` vs `fmt.Printf("text")`

Reply:
Oh, I see! A subtle bug, good catch.